<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1081</width>
    <height>830</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_12">
    <item>
     <widget class="QLabel" name="txt_title">
      <property name="font">
       <font>
        <family>Broadway</family>
        <pointsize>28</pointsize>
       </font>
      </property>
      <property name="text">
       <string>MoonShineKedit</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_11">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="label_16">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Camera</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_17">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Realtime filter</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QGraphicsView" name="cmr_org"/>
          </item>
          <item>
           <widget class="QGraphicsView" name="cmr_out"/>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label_14">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Original Image</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_15">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Generated Image</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QGraphicsView" name="pht_org">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QGraphicsView" name="pgt_new"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_4"/>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QPushButton" name="btn_loadimg">
            <property name="text">
             <string>Load Image</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_generateimg">
            <property name="text">
             <string>Filter</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <layout class="QGridLayout" name="gridLayout">
                <item row="2" column="0">
                 <widget class="QLabel" name="label_5">
                  <property name="text">
                   <string>Threshold Value </string>
                  </property>
                 </widget>
                </item>
                <item row="6" column="1">
                 <widget class="QLineEdit" name="thr_c"/>
                </item>
                <item row="7" column="0">
                 <widget class="QCheckBox" name="thr_otsu">
                  <property name="font">
                   <font>
                    <pointsize>11</pointsize>
                   </font>
                  </property>
                  <property name="layoutDirection">
                   <enum>Qt::LeftToRight</enum>
                  </property>
                  <property name="text">
                   <string>3- Otsu thresholding</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="label">
                  <property name="font">
                   <font>
                    <pointsize>11</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>2- Adaptive Thresholding</string>
                  </property>
                 </widget>
                </item>
                <item row="5" column="1">
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>C</string>
                  </property>
                 </widget>
                </item>
                <item row="6" column="0">
                 <widget class="QLineEdit" name="thr_block_size"/>
                </item>
                <item row="3" column="0">
                 <widget class="QLineEdit" name="thr_value"/>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_4">
                  <property name="font">
                   <font>
                    <pointsize>11</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>1- Simple Thresholding</string>
                  </property>
                 </widget>
                </item>
                <item row="5" column="0">
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>Block size</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="txt_thresholding">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Thresholding</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <layout class="QGridLayout" name="gridLayout_3">
                <item row="0" column="0">
                 <widget class="QLabel" name="txt_blur">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Blur</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_10">
                  <property name="text">
                   <string>Kernel size</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="bl_kernelsize"/>
                </item>
                <item row="1" column="2">
                 <widget class="QCheckBox" name="bl_filter">
                  <property name="text">
                   <string>Box Filter</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="3">
                 <widget class="QCheckBox" name="bl_gaussian">
                  <property name="text">
                   <string>Gaussian blur</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QCheckBox" name="bl_normal">
                  <property name="text">
                   <string>Normal blur</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="2">
                 <widget class="QCheckBox" name="bl_bilateral">
                  <property name="text">
                   <string>Bilateral fiter</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="3">
                 <widget class="QCheckBox" name="bl_median">
                  <property name="text">
                   <string>Median blur</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_3">
                <item>
                 <widget class="QLabel" name="txt_border">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                    <kerning>true</kerning>
                   </font>
                  </property>
                  <property name="text">
                   <string>Border</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <layout class="QGridLayout" name="gridLayout_2">
                  <item row="1" column="3">
                   <widget class="QLineEdit" name="brd_w"/>
                  </item>
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_6">
                    <property name="text">
                     <string>Red</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="0" colspan="3">
                   <widget class="QCheckBox" name="brd_replicate">
                    <property name="text">
                     <string>Border constant</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QLabel" name="label_7">
                    <property name="text">
                     <string>Blue</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="3">
                   <widget class="QLabel" name="label_9">
                    <property name="text">
                     <string>Border width</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="2">
                   <widget class="QLineEdit" name="brd_g"/>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLineEdit" name="brd_r"/>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLineEdit" name="brd_b"/>
                  </item>
                  <item row="0" column="2">
                   <widget class="QLabel" name="label_8">
                    <property name="text">
                     <string>Green</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="2" column="0">
               <widget class="QCheckBox" name="bda_sobel">
                <property name="text">
                 <string>Sobel</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QCheckBox" name="bda_laplacian">
                <property name="text">
                 <string>Laplacian</string>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QCheckBox" name="bda_canny">
                <property name="text">
                 <string>Canny</string>
                </property>
               </widget>
              </item>
              <item row="2" column="3">
               <widget class="QCheckBox" name="bda_deriche">
                <property name="text">
                 <string>Deriche</string>
                </property>
               </widget>
              </item>
              <item row="2" column="4">
               <widget class="QCheckBox" name="bda_harris">
                <property name="text">
                 <string>Harris</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0" colspan="5">
               <widget class="QLabel" name="txt_borderdetectionalgorithms">
                <property name="font">
                 <font>
                  <pointsize>14</pointsize>
                 </font>
                </property>
                <property name="text">
                 <string>Border Detection Algorithms</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QGridLayout" name="gridLayout_5">
              <item row="1" column="0" colspan="2">
               <widget class="QLineEdit" name="gmm_value"/>
              </item>
              <item row="0" column="0" colspan="2">
               <widget class="QLabel" name="txt_gammavalue">
                <property name="font">
                 <font>
                  <pointsize>14</pointsize>
                 </font>
                </property>
                <property name="text">
                 <string>Gamma value</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QCheckBox" name="gmm_sharpining">
                <property name="text">
                 <string>Sharpining</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QCheckBox" name="gmm_outline">
                <property name="text">
                 <string>Outline</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QLabel" name="label_11">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Face Detection</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="label_12">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Countur Detection</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_13">
              <property name="font">
               <font>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Watershed Algorithm</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QCheckBox" name="dt_viola">
              <property name="text">
               <string>Viola jones Haar Cascade</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QCheckBox" name="dt_kontur">
              <property name="text">
               <string>Countur</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QCheckBox" name="dt_watershed">
              <property name="text">
               <string>Watershed</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="btn_reset">
            <property name="text">
             <string>Reset Parameters</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_savepar">
            <property name="text">
             <string>Save Parameters</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_loadpar">
            <property name="text">
             <string>Load Parameters</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_saveimg">
            <property name="text">
             <string>Save New Image</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_exit">
            <property name="text">
             <string>Exit</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1081</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="mnbar_help">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actbtn_about"/>
   </widget>
   <addaction name="mnbar_help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actbtn_about">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
